### DISCUSSION TITLE
Is it possible to reject a Model that has two identical fields?

### DISCUSSION THREAD

**Discussion by MoGuGuai-hzr** [pydantic/pydantic#8941] [Question] [answered]
In the current scenario, if the same field appears multiple times, the latter one will overwrite the previous one. For example, the following code will output `x=2`.

```python
from pydantic import BaseModel


class Params(BaseModel):
    x: int = 1
    x: int = 2


h = Params()
print(h)

```

Is there a way to prohibit the same fields? In the case of many parameters, I may not notice it. The previous value is overwritten by the later value.

**Comment by Viicos** [comment]
Could be done in a pretty straightforward way by tweaking the namespace dict returned by the model metaclass' `__prepare__`, but this would only take into account assignments (and not bare annotations like `x: int`, without a default value -- a least without hacking into the `__annotations__` dict).

However, I don't know if maintainers would be ok to have it as it is a pretty niche use case. I'll probably add it to `flake8-pydantic`, a plugin I created a couple weeks ago.

**Reply by MoGuGuai-hzr** [reply]
Looking forward to your library

**Comment by Viicos** [comment, marked as answer]
I just published [`flake8-pydantic`](https://github.com/Viicos/flake8-pydantic) 0.3.0, which includes a rule for this.

**Reply by MoGuGuai-hzr** [reply]
nice
